mssqlscripter/mssqltoolsservice/external.py

```
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import print_function
+
+import os
+import tarfile
+import utility
+import zipfile
+
+from future.standard_library import install_aliases
+install_aliases()
+
+
+SQLTOOLSSERVICE_BASE = os.path.join(utility.ROOT_DIR, 'sqltoolsservice/')
+
+# Supported platform key's must match those in mssqlscript's setup.py.
+SUPPORTED_PLATFORMS = {
+    'manylinux1_x86_64': SQLTOOLSSERVICE_BASE + 'manylinux1/' + 'Microsoft.SqlTools.ServiceLayer-linux-x64-netcoreapp2.0.tar.gz',
+    'manylinux1_i686': SQLTOOLSSERVICE_BASE + 'manylinux1/' + 'Microsoft.SqlTools.ServiceLayer-linux-x64-netcoreapp2.0.tar.gz',
+    'macosx_10_11_intel': SQLTOOLSSERVICE_BASE + 'macosx_10_11_intel/' + 'Microsoft.SqlTools.ServiceLayer-osx-x64-netcoreapp2.0.tar.gz',
+    'win_amd64': SQLTOOLSSERVICE_BASE + 'win64/' + 'Microsoft.SqlTools.ServiceLayer-win-x64-netcoreapp2.0.zip',
+    'win_64': SQLTOOLSSERVICE_BASE + 'win64/' + 'Microsoft.SqlTools.ServiceLayer-win-x64-netcoreapp2.0.zip',
+    'win32': SQLTOOLSSERVICE_BASE + 'win32/' + 'Microsoft.SqlTools.ServiceLayer-win-x86-netcoreapp2.0.zip'
+}
+
+TARGET_DIRECTORY = os.path.abspath(os.path.join(os.path.abspath(__file__), '..', 'bin'))
+
+
+def copy_sqltoolsservice(platform):
+    """
+        For each supported platform, build a universal wheel.
+    """
+    # Clean up dangling directories if previous run was interrupted.
+    utility.clean_up(directory=TARGET_DIRECTORY)
+
+    if not platform or platform not in SUPPORTED_PLATFORMS:
+        print('Please provide a valid platform flag.' +
+              '[win32, win_amd64, win64, manylinux1_x86_64, manylinux1_i686, macosx_10_11_intel]')
+        return
```

>return [](start = 8, length = 6)

we probably should error out here, rather than continuing on.